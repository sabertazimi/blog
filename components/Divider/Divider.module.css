.divider.divider {
  @apply leading-relaxed tracking-wide;
}

.divider.divider[role='separator'] {
  @apply mx-0 my-12 font-extrabold;
}
