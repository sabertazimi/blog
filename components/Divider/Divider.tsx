import type { DividerProps } from 'antd'
import { Divider as AntDivider } from 'antd'
import cx from 'classnames'
import styles from './Divider.module.css'

interface Props extends DividerProps {}

function Divider({
  type = 'horizontal',
  className,
  children,
}: Props): JSX.Element {
  return (
    <AntDivider
      type={type}
      className={cx(
        styles.divider,
        className,
        'dark:!border-light dark:text-light',
      )}
    >
      {children}
    </AntDivider>
  )
}

export default Divider
