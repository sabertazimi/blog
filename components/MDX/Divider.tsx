import Divider from '@components/Divider';

const MDXDivider = (): JSX.Element => (
  <Divider className="mx-0 my-12 h-1 bg-gradient-primary" />
);

export default MDXDivider;
