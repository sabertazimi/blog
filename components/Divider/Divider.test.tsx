import { render } from '@testing-library/react';
import Divider from './Divider';

describe('Divider', () => {
  test('should render correctly (snapshot)', () => {
    const { container } = render(<Divider>Divider</Divider>);

    expect(container).toMatchSnapshot();
  });
});
