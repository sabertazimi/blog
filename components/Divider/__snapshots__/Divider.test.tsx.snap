// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render correctly (snapshot) 1`] = `
<div>
  <div
    class="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center mx-0 my-12 font-extrabold"
    role="separator"
  >
    <span
      class="ant-divider-inner-text"
    >
      Divider
    </span>
  </div>
</div>
`;
